### PR TITLE
Conner's RN DevTools Documentation

### PR DESCRIPTION
**Pull request by ConnorLand**
I added documentation written by @cdcharlebois to our existing native debug page. Posting here for quick technical checks before merging.

### CODE REVIEW

File: content/howto/mobile/native-debug.md [hunk 1]
```diff
@@ -39,7 +39,38 @@ If the status remains at **Waiting**, use the reload command (pictured above) to
...
+
+### 2.1 Using React Developer Tools{#rn-dev}
+
+React Developer Tools is an extension for [Chrome](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en) and [Firefox](https://addons.mozilla.org/en-US/firefox/addon/react-devtools/) browers which will allow you to see investigate the way your native page is rendering, adjust things like spacing in a live editor, and inspect the state and props of your pluggable and native widgets. You can consult Facebook's [official documentation](https://github.com/facebook/react/tree/master/packages/react-devtools) for extra information, but this document will teach you the basics of using React DevTools.
+
+To install React Developer Tools, do the following:
+
+1. Open your CLI and run NPX (an executable runner for NPM) with this code: `npx react-devtools@^3`. The `@^3` ensures compatibility with Mendix's React Native version.
+
+#### 2.1.1 Debugging with a Simulator
```
Comment (ConnorLand) [inline content/howto/mobile/native-debug.md:52]: We use "emulator" elsewhere in the docs. I think an emulator is what's required here, right?

File: content/howto/mobile/native-debug.md [hunk 2]
```diff
@@ -39,7 +39,38 @@ If the status remains at **Waiting**, use the reload command (pictured above) to
...
+
+Open your native app in your simulator of choice then do the following:
```
Comment (ConnorLand) [inline content/howto/mobile/native-debug.md:54]: Do we want to recommend a simulator/emulator here with a link?

File: content/howto/mobile/native-debug.md (hunk 1 above)
Comment (cdcharlebois) [reply content/howto/mobile/native-debug.md:52]: I would think that "emulator" is fine; I just haven't tested it with the Android Studio emulator--only the XCode one.

File: content/howto/mobile/native-debug.md (hunk 2 above)
Comment (cdcharlebois) [reply content/howto/mobile/native-debug.md:54]: I just used the simulator built into XCode. I think the only other emulator we'd suggest would be the one built into Android Studio, right?

Comment (ConnorLand) [reply content/howto/mobile/native-debug.md:54]: Oh right, capital S Simulator. Ok thanks Conner that clears it up. I'll change all the iOS simulator mentions to Simulator. Thanks.

File: content/howto/mobile/native-debug.md [hunk 3]
```diff
@@ -39,7 +39,38 @@ If the status remains at **Waiting**, use the reload command (pictured above) to
...
 
-Outside of Chrome, other tools can help you debug Mendix apps (such as React Developer Tools). Regardless of which tool you use, remember that Mendix uses a different port (8083) than a default React Native installation would (8080).
+Other tools can help you debug Mendix apps, such as the [Using React Developer Tools](#rn-dev) section below. Regardless of which tool you use, remember that Mendix uses a different port (8083) than a default React Native installation would (8080).
+
+### 2.1 Using React Developer Tools{#rn-dev}
+
+React Developer Tools is an extension for [Chrome](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en) and [Firefox](https://addons.mozilla.org/en-US/firefox/addon/react-devtools/) browsers which will allow you to see investigate the way your native page is rendering, adjust things like spacing in a live editor, and inspect the state and props of your pluggable and native widgets. You can consult Facebook's [official documentation](https://github.com/facebook/react/tree/master/packages/react-devtools) for extra information, but this document will teach you the basics of using React Developer Tools.
```
Comment (smarki) [inline content/howto/mobile/native-debug.md:46]: You are referring to the browser extension but the document explains how to use the standalone app; is that correct?

Comment (smarki) [reply content/howto/mobile/native-debug.md:46]: The link for the app itself is https://github.com/facebook/react/tree/master/packages/react-devtools

Comment (smarki) [reply content/howto/mobile/native-debug.md:46]: In addition a user would have to have Node & NPM installed to be able to run the `npx` command. Maybe we should mention it as a requirement.

File: content/howto/mobile/native-debug.md [hunk 4]
```diff
@@ -39,7 +39,38 @@ If the status remains at **Waiting**, use the reload command (pictured above) to
...
+
+To install React Developer Tools, do the following:
+
+1. Open your CLI and run NPX (an executable runner for NPM) with this code: `npx react-devtools@^3`. The `@^3` ensures compatibility with Mendix's React Native version.
+
+#### 2.1.1 Debugging with iOS Simulator
```
Comment (smarki) [inline content/howto/mobile/native-debug.md:52]: This is the same for an android simulator too.

File: content/howto/mobile/native-debug.md [hunk 5]
```diff
@@ -39,7 +39,38 @@ If the status remains at **Waiting**, use the reload command (pictured above) to
...
+
+Open your native app in iOS Simulator and then do the following:
+
+1. Select **Enable dev mode** on your native app.
+2. Run `npx react-devtools@^3`.
+3.  React Developer Tools will launch and connect to Simulator. You can now inspect and modify the React Native elements the same way you could modify HTML elements in Chrome:
+
```
Comment (smarki) [inline content/howto/mobile/native-debug.md:59]: a 4. step should be to `Toggle element inspector` for the dev menu (3 finger tap) to enable the enhanced inspecting functionality

File: content/howto/mobile/native-debug.md [hunk 6]
```diff
@@ -39,7 +39,38 @@ If the status remains at **Waiting**, use the reload command (pictured above) to
...
+	{{% image_container width="350" %}}![debug simulator](attachments/native-debug/simulator-rn-dev.png){{% /image_container %}}
+
+#### 2.1.2 Debugging with the Make It Native App
+
+To use the Make It Native app with React Developer Tools, do the following: 
+
+1. Connect your mobile device to your laptop with a USB cord.
+2. Run `adb devices` to ensure your device is listed.
+3. Start your native app on your device with **Enable dev mode** selected.
```
Comment (smarki) [inline content/howto/mobile/native-debug.md:68]: a 4. step should be to `Toggle element inspector` for the dev menu (3 finger tap) to enable the enhanced inspecting functionality

File: content/howto/mobile/native-debug.md (hunk 3 above)
Comment (ConnorLand) [reply content/howto/mobile/native-debug.md:46]: Ok I've committed these changes

File: content/howto/mobile/native-debug.md (hunk 4 above)
Comment (ConnorLand) [reply content/howto/mobile/native-debug.md:52]: incorporated this comment

File: content/howto/mobile/native-debug.md (hunk 6 above)
Comment (ConnorLand) [reply content/howto/mobile/native-debug.md:68]: incorporated this comment

**Pull request merged**